utils/io_utils.py

```diff
@@ -0,0 +1,56 @@
+"""
+This module provides functions for loading, writing, and modifying files in this data 
+repository. It centralizes input/output operations, ensuring efficient and consistent 
+handling of data files throughout the project.
+"""
+import pathlib
+from collections import defaultdict
+import pandas as pd
+
+def load_barcodes(barcode_path: str | pathlib.Path) -> dict:
```

Comment:
Looks like this `.py` file needs a quick autoformatting tool applied? (e.g., ruff)? (there should be two new line characters before function definitions, I believe)